homeassistant/components/sensor/fastdotcom.py

```diff
@@ -78,22 +78,13 @@ def update(self):
         """Get the latest data and update the states."""
         data = self.speedtest_client.data
         if data is None:
-            entity_id = 'sensor.fastcom_speedtest'
-            states = recorder.get_model('States')
-            try:
-                last_state = recorder.execute(
-                    recorder.query('States').filter(
-                        (states.entity_id == entity_id) &
-                        (states.last_changed == states.last_updated) &
-                        (states.state != 'unknown')
-                    ).order_by(states.state_id.desc()).limit(1))
-            except TypeError:
-                return
-            except RuntimeError:
+            if self._state is not None:
```

Comment:
Let's keep any integration into other components out of the initial release. Let's really focus on the core support.

Because if we do it right, components/platforms don't have to restore anything. They can just not set a state and then restoration will do it instead .